todo/src/App.js

```diff
@@ -0,0 +1,53 @@
+import React, {Component} from 'react';
+import './App.css';
+import TodoList from './components/TodoList';
+import TodoForm from './components/TodoForm';
+import { connect } from 'react-redux';
+import { addTodo } from './actions';
+
+
+ class App extends Component {
+   constructor(props){
+     super(props);
+   this.state = {
+     textHolder: ''
+   }
+  }
+
+  handleInputChange = (event) => {
+   this.setState( { [event.target.name]: event.target.value });
+  }
+
+  addTodo = () => {
+    let todoText = this.state.textHolder;
+    let newTodo = {
+      id: this.props.todos.length,
+      completed: false,
+      text: todoText
+    };
+    this.props.addTodoHandler(newTodo);
+    this.setState( { textHolder: ''} );
+  }
+
+   render () {
+     return (
+       <div className="App">
+         <TodoList todos={this.props.todos} />
+          <TodoForm handleInputChange={this.handleInputChange} textHolder={this.state.textHolder} addTodo={this.addTodo} />
```

Comment:
Instead of passing down all of these props, use the necessary functions (like `mapStateToProps` or `connect`) inside of the `TodoForm` component.